pandas/core/dtypes/cast.py

```diff
@@ -60,6 +62,24 @@
 _int16_max = np.iinfo(np.int16).max
 _int32_max = np.iinfo(np.int32).max
 _int64_max = np.iinfo(np.int64).max
+_int64_min = np.iinfo(np.int64).min
+_uint64_max = np.iinfo(np.uint64).max
+_float32_max = np.finfo(np.float32).max
+
+
+def _is_iNaT(x):
+    """
+    Helper function to circumvent numpy bug for timedeltas
+
+    Specifically, comparing a scalar timedelta against another scalar value may
+    raise a spurious DeprecationWarning, see numpy/numpy#10095
+    """
+    if not is_scalar(x):
```

Comment:
check is_integer and avoid the need for the filterwarning (i think is_integer behavior may have been changed since this was writtten)

Reply:
OK, I chased down `is_integer` to https://github.com/pandas-dev/pandas/blob/0bde7cedf46209a9fd4fa8c7f9fbce8b49aa78cd/pandas/_libs/tslibs/util.pxd#L79 and that excludes timedeltas now. Will give it a try.